TWLight/applications/models.py

```diff
@@ -319,16 +318,24 @@ def update_app_status_on_save(sender, instance, **kwargs):
             instance.date_closed = localtime(now()).date()
             instance.days_open = 0
 
-# Authorize editor to access resource after an application is saved as approved
-# or sent. We need to use post_revision_commit from django-reversion to get the
-# **current** revision user. Just listening to the post_save signal will grab
-# the user from the previously saved version.
+# Authorize editor to access resource after an application is saved as sent.
 @receiver(post_save, sender=Application)
 def post_revision_commit(sender, instance, **kwargs):
-    if reversion.is_active() and (instance.status == 2 or instance.status == 4) and not instance.imported:
+    # Check if an authorization already exists.
+    if instance.specific_stream:
+        existing_authorization = Authorization.objects.filter(
+            authorized_user=instance.user,
+            partner=instance.partner,
+            stream=instance.specific_stream).exists()
+    else:
+        existing_authorization = Authorization.objects.filter(
+            authorized_user=instance.user,
+            partner=instance.partner).exists()
+
+    if instance.status == Application.SENT and not instance.imported:
```

Comment:
Should we be using the `existing_authorization` in our guard clause? something like this untested code?
```suggestion
    # Create an authorization if this is a newly finalized application for a user that isn't already authrorized.
    if instance.status == Application.SENT and not instance.imported and not existing_authorization:
```